### ISSUE TITLE
please  

### ISSUE THREAD

**Issue by 116705792** [pumpikano/tf-dann#19]
hey,I have some questions. 
firstly,in your code :  total_loss=domain_loss + predict_loss.
but in the paper total_loss= domain_loss -  lamda*predict_loss.
secondly,this is because GRL layer?

**Comment by jindongwang** [comment]
I have these questions, too.

**Comment by Engineero** [comment]
@116705792 other way around. The paper is finding a saddle point of the objective function `E = predict_loss - lambda*domain_loss` (equation 9 in [Ganin, 2016](https://arxiv.org/abs/1505.07818)) by finding the parameters that *minimize* `predict_loss` and *maximize* `domain_loss`. Maximizing is the same as minimizing the inverse, so this is identical to finding the parameters that jointly minimize `predict_loss + lambda*domain_loss`. The `lambda` is implemented as `l` in the gradient reversal layer, flip_gradient.py.

**Comment by 116705792** [comment]
thank  you  very much! i m learn a lot
